helpers/image/image.go

```diff
@@ -50,7 +51,7 @@ func NewReference(name string) (*Reference, error) {
 	}
 
 	// Get the hostname
-	hostname, _ := reference.SplitHostname(ref)
+	hostname, reponame := reference.SplitHostname(ref)
```

Comment:
SplitHostname is deprecated function, better to use latest github.com/docker/distribution/reference (v2.7.1+incompatible) and functions Domain and Path. Then you will get also the port.
hostname := reference.Domain(ref)
reponame := reference.Path(ref)

also the GetHostname() function is used only for querying notary and should contain port, if specified.
func (r Reference) GetHostname() string {
       port := r.port
       if port != "" {
           port = ":" + port
       }
       return r.hostname + port
}

Reply:
Thanks for the review @rubeoboletus! I will refactor this in #82